### ISSUE TITLE
Inconsistency in handing buildscript block nested inside subprojects block

### ISSUE THREAD

**Issue by JLLeitschuh** [gradle/kotlin-dsl#374]
This bit of code in the parent project behaves correctly:
```kotlin
subprojects {
    buildscript.repositories {
        mavenCentral()
    }
}
```
I would expect this to work as well, but it does not correctly configure the `subprojects`.
```kotlin
subprojects {
    buildscript {
        repositories {
            mavenCentral()
        }
    }
}
```

Another side note, this won't even compile.
```kotlin
subprojects {
    this.buildscript {
        repositories {
            mavenCentral()
        }
    }
}
```
Version:
```
distributionUrl=https\://services.gradle.org/distributions/gradle-3.5-all.zip
```

**Comment by bamboo** [comment]
This also applies to init scripts:

```kotlin
// ~/.gradle/init.d/local-repositories.init.gradle.kts
projectsLoaded {
  rootProject.buildscript {
     repositories { mavenLocal() }
  }
}
```

**Comment by eskatos** [comment]
```kotlin
fun Project.buildscript(action: ScriptHandlerScope.() -> Unit): Unit =
    action(ScriptHandlerScope((project as ProjectInternal).buildscript))
```

**Comment by eskatos** [comment]
Closing as fixed in 651248f00cd02b2ae0200b2ff1d5979e3e2de891
